upgrade.js

```
@@ -1,397 +1,153 @@
-      installedFrameworkVersion = versionFile.adapt_framework;
-    }
-
-    console.log('Currently installed versions:\n- ' + app.polyglot.t('app.productname') + ': ' + installedBuilderVersion + '\n- Adapt Framework: ' + installedFrameworkVersion);
-    callback();
-
-  },
-  function(callback) {
+var configuration = require('./lib/configuration');
+var logger = require('./lib/logger');
+var origin = require('./lib/application');
+var OutputConstants = require('./lib/outputmanager').Constants;
+var installHelpers = require('./lib/installHelpers');
 
-    console.log('Checking for ' + app.polyglot.t('app.productname') + ' upgrades...');
-    // Check the latest version of the project
-    request({
-      headers: {
-        'User-Agent' : DEFAULT_USER_AGENT
-      },
-      uri: 'https://api.github.com/repos/adaptlearning/adapt_authoring/tags',
-      method: 'GET'
-    }, function (error, response, body) {
-
-      if (!error && response.statusCode == 200) {
-        var tagInfo = JSON.parse(body);
+var DEFAULT_USER_AGENT = 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36';
+var app = origin();
 
-        if (tagInfo) {
-          latestBuilderTag = tagInfo[0].name;
-        }
+/**
+* Start of execution
+*/
+start();
+
+function start() {
+  // don't show any logger messages in the console
+  logger.level('console','error');
+  // start the server first
+  app.run({ skipVersionCheck: true, skipStartLog: true });
+  app.on('serverStarted', getUserInput);
+}
 
-        callback();
+function getUserInput() {
+  // properties for the prompts
+  var confirmProperties = {
+    name: 'continue',
+    description: 'Continue? Y/n',
+    type: 'string',
+    default: 'Y',
+    before: installHelpers.inputHelpers.toBoolean
+  };
+  var upgradeProperties = {
+    properties: {
+      updateAutomatically: {
+        description: 'Update automatically? Y/n',
```

If I'm reading the [code](https://github.com/taylortom/adapt_authoring/blob/feature/upgrade/lib/installHelpers.js#L190) correctly automatic update checks versions from the adapt github repo. This may not be the expected behaviour is the user has previously specified a different repo in config/install. May just need a wording change here to advise, otherwise working really well.

---

If you select an automatic update and have different repos defined it will just exit with an error saying It cant do that. At least it did in my original code.

---

Apologies, your right, should have tested before commenting. Gives error like this:
> Update automatically? Y/n (Y) 
Software updates found.
Failed to upgrade /temp/59d4aadeb458b3861c90c495/adapt_framework to v2.2.2
ERROR: error: pathspec 'v2.2.2' did not match any file(s) known to git.
Upgrade was unsuccessful. Please check the console output.

---

Then something is broken it should be stopping you doing an auto upgrade with custom repos.

---

Issue's caused by [lib/installHelpers.js#L190](https://github.com/taylortom/adapt_authoring/blob/06d08e49162c2a889777f4d065a78f01153331cc/lib/installHelpers.js#L190).

I'm not sure that I agree that the script should just throw an error if custom repos are used. It could be useful to be able to perform an auto-upgrade, and there's no technical reason we can't (obviously the tags would need to follow semver for the upgrades to work...).

---

At least for our use case, the tags don't follow I doubt most people will that kind of the point of using your own repo. I think is someone has gone to the trouble of setting up custom repos they probably know enough to and want to set what version they want.

I don't see this as a feature anyone but the power users will use.